### ISSUE TITLE
Support for Secure Payment Confirmation

### ISSUE THREAD

**Issue by benfoster** [passwordless-lib/fido2-net-lib#267]
[Secure Payment Confirmation] builds on top of WebAuthN and uses the Payment Request API to allow payments to be authenticated.

I've been able to adapt the demo application to use SPC but unfortunately the verification of the assertion fails because the type is no longer `webauthn.get` but `payment.get` (see [AuthenticatorResponse](https://github.com/passwordless-lib/fido2-net-lib/blob/25a30130f4b4cbd73f11165a4a14055a49c511bb/Src/Fido2/AuthenticatorResponse.cs#L64)).

Is it possible to override this verification or add support for additional client data types?

**Comment by aseigler** [comment]
Yeah, looks completely doable, and the architects built it to be so compatible by design.

**Comment by thomasnymand** [comment]
I have also stumbled on this issue. Any new considerations regarding possible override option or workaround?

**Comment by abergs** [comment]
I haven't paid too much attention to SPC. If someone is interested in doing a PoC Draft PR to show how we would implement this in a clean way I'd be interested in reviewing it.
